app/workers/commit_monitor_handlers/commit_range/github_pr_commenter/commit_metadata_checker.rb

```diff
@@ -0,0 +1,63 @@
+require 'rugged'
+
+module CommitMonitorHandlers::CommitRange
+  class GithubPrCommenter::CommitMetadataChecker
+    include Sidekiq::Worker
+    sidekiq_options :queue => :miq_bot_glacial
+
+    include BatchEntryWorkerMixin
+    include BranchWorkerMixin
+
+    def perform(batch_entry_id, branch_id, new_commits)
+      return unless find_batch_entry(batch_entry_id)
+      return skip_batch_entry unless find_branch(branch_id, :pr)
+
+      complete_batch_entry(:result => process_commits(new_commits))
+    end
+
+    private
+
+    def process_commits(new_commits)
+      @offenses = []
+
+      new_commits.each do |commit_sha, data|
+        check_for_usernames_in(commit_sha, data["message"])
+      end
+
+      @offenses
```

Comment:
This feels a bit awkward that `@offenses` is created and returned here, but `check_for_usernames_in` is the only one that adds to the array.  Would it be better to `new_commits.flat_map` instead?

Reply:
@bdunne so, wanted to double check this, but this is pretty much the standard for the other two workers are doing things:

https://github.com/ManageIQ/miq_bot/blob/903eef83bb8404a7f7d8ff782ebb81c08ba68498/app/workers/commit_monitor_handlers/commit_range/github_pr_commenter/diff_filename_checker.rb#L20-L26
https://github.com/ManageIQ/miq_bot/blob/903eef83bb8404a7f7d8ff782ebb81c08ba68498/app/workers/commit_monitor_handlers/commit_range/github_pr_commenter/diff_content_checker.rb#L20-L26

So didn't really want break from convention for this one.

Do you mind if we address this in a follow up?

Reply:
> Do you mind if we address this in a follow up?

:+1: Fine by me